### ISSUE TITLE
PiHole dual operation (for remote devices & local network)

### ISSUE THREAD

**Issue by zilexa** [mr-bolle/docker-openvpn-pihole#18]
Question, not an issue:
Is this possible with your image? 
It is also described here: https://docs.pi-hole.net/guides/vpn/dual-operation/

Your diagram visualises almost exactly what I would like. I have a Intel/Ubuntu system running docker images already, adding your image would be easy for the road warrior setup.

However I would like to set the PiHole DNS as the DNS server in my home router as well. Is that possible?

I noticed in a closed issue, you recommended someone to install a secondary instance of PiHole. That's really not what I want since I will whitelist some IPs/addresses over time and don't want to manage that on 2 different PiHoles. 

Would the documentation mentioned above be applicable to your image?

**Comment by mr-bolle** [comment]
Hi, this setup is for a VPN connection with Pi-Hole as DNS server. 
Why don't you use a separate pi-hole container as a local DNS server?

**Comment by zilexa** [comment]
Well, that's the reverse of my question and I asked first ;) 
To answer your question: I prefer to run 1 instance no need to use up extra resources. Also, when I whitelist a site or domain I will have to go through the hassle of whitelisting it on both instances.

**Comment by mr-bolle** [comment]
Unfortunately, I cannot give you any tips on how to set up this setup.  So far I have had no need for this, the need for resources is not a problem for me.  Maybe you can find it at https://github.com/pi-hole/docker-pi-hole/

**Comment by EmpressFiona** [comment]
For the record, doing this is as simple as opening up the `docker-compose.yml` file, and adding in two ports under pi-hole as follows:

```
     ports:
       #- "553:53/tcp"
       #- "553:53/udp"
       - "53:53/tcp"
       - "53:53/udp"
       - "8081:80/tcp"
```

Also, if you've configured your own custom DNS servers (for instance, if you're using conditional forwarding to your router), comment out the following (and use them as your router's resolver if you want), or they'll reset whenever the container has to restart:

```
# #1 Digitalcourage | #2 Chaos Computer Club
#       DNS1: 46.182.19.48
#       DNS2: 213.73.91.35
```

Then type `docker-compose up -d` and reload the container.

(There may be issues I haven't anticipated with this, however, it's working fine like this for me right now.)

**Comment by zilexa** [comment]
@EmpressFiona actually no that did not work. I have tried A LOT of things including that, but also played with macvlans. As long as Pihole ran in Docker and my (PiVPN-Wireguard) bare, Pihole was not reachable when connected to Wireguard. 

I switched to AdGuard Home, it is a single binary + config file: although running via Docker is possible, it would not make sense so I simply run it bare and everything works OK now with Wireguard VPN. 

PiHole is really a bunch of 3rd party (not developed by PiHole) tools held together with a whole lot of PHP. AdGuard Home is a much cleaner solution in my opinion.
